nltk/metrics/distance.py

```diff
@@ -182,18 +186,62 @@ def custom_distance(file):
             labelA, labelB, dist = l.strip().split("\t")
             labelA = frozenset([labelA])
             labelB = frozenset([labelB])
-            data[frozenset([labelA,labelB])] = float(dist)
-    return lambda x,y:data[frozenset([x,y])]
+            data[frozenset([labelA, labelB])] = float(dist)
+    return lambda x, y: data[frozenset([x, y])]
+
+
+def jaro_winkler_distance(s1, s2, p=0.1):
+    """ Details about this implementation can be found at
+    https://en.wikipedia.org/wiki/Jaro-Winkler_distance
+dist_bound = type int, upper bound of distance for being a matched character
+mtch = type int, no.of matched characters in s1 and s2
+t_cnt= half of the number of transpositions between s1 and s2
+j_sim = type float ,jaro similarity between s1 and s2
+p = type float , weight parameter with a standard value of 0.1
+l_cnt = type int, common prefix at the start of the string (max val = 4)
+jw_sim= type float, jaro winkler similarity between s1 and s2
+jw_dist = type float, it is equal to 1 - jw_sim """
+    s1 = s1.lower()
+    s2 = s2.lower()
+    mtch = 0
+    dist_bound = np.floor(max(len(s1), len(s2)) / 2) - 1
+    t_cnt = 0
+    l_cnt = 0
+    for ch1 in s1:
+        if ch1 in s2:
+            pos1 = s1.index(ch1)
+            pos2 = s2.index(ch1)
+            if(abs(pos1 - pos2) <= dist_bound):
+                mtch = mtch + 1
+                if(pos1 != pos2):
+                    t_cnt = t_cnt + 1
+    t_cnt = t_cnt // 2
+    if mtch == 0:
+        j_sim = 0
+    else:
+        j_sim = (mtch / len(s1) + mtch / len(s2) + (mtch - t_cnt) / mtch) / 3
+    for i in range(len(s1)):
+        if(s1[i] == s2[i]):
+            l_cnt = l_cnt + 1
+        else:
+            break
+        if(l_cnt == 4):
+            break
```

Comment:
These conditions checks don't exactly makes use of `numpy`. I think there are better ways to do the match.

Reply:
Actually this look like RIBES' https://github.com/nltk/nltk/blob/develop/nltk/translate/ribes_score.py#L233 I might be wrong.

Reply:
Well i actually used numpy for the np.floor(). Also I wen through RIBES and the find_increasing_sequences(worder) function actually returns monotonic sequences given a worder list. But l_cnt here is actually this :  the length of common prefix at the start of the string up to a maximum of four characters .  It is as per this:  https://en.wikipedia.org/wiki/Jaro%E2%80%93Winkler_distance